dataprofiler/profilers/float_column_profile.py

```diff
@@ -106,16 +106,26 @@ def _get_float_precision(cls, df_series_clean):
         :return: string representing its precision print format
         :rtype: int
         """
-        if not len(df_series_clean):
-            return 0
+        if not len(df_series_clean): return 0
 
         # Lead zeros: ^[+-.0\s]+ End zeros: \.?0+(\s|$)
         # Scientific Notation: (?<=[e])(.*) Any non-digits: \D
         r = re.compile(r'^[+-.0\s]+|\.?0+(\s|$)|(?<=[e])(.*)|\D')
-        
-        return float(df_series_clean.replace(
-            to_replace=r, value='').map(len).min())
-        
+
+        # Take a sampling of the dataset. If small use full dataset,
+        # OR 20k samples or 5% of the dataset which ever is larger.
+        sample_size = min(len(df_series_clean),
```

Comment:
The concern i have is this will not reflect the sample size given to the user in the profile. If this does happen, we should at least warn the user that the sampling provide by the profile is not the same for precision calculations

Reply:
Proposal, move precision to a section in the dict & make an option enabling the user to set their desired ratio of sampling.

```
precision: {
    'min': 0,
    'max': 5,
    'mean': 4.23,
    'std': 0.1,
    'samples': 20000,
    'confidence_level': 99.99%
}
```
In terms of what that would look like for average precision with 99.9% confidence level:

`4.23 ± 0.00233 = 4.23 +/- 3.291*(0.1/sqrt(200000))`

https://www.calculator.net/confidence-interval-calculator.html?size=20000&mean=4.3&sd=3.2&cl=95&x=94&y=18

Reply:
```
# Take a sampling of the dataset. If small use full dataset,                       
# OR 20k samples or 5% of the dataset which ever is larger.                        
sample_size = min(len(df_series_clean), max(20000, int(len(df_series_clean)/20)))

# length of sampled cells after all punctuation removed                            
len_per_float = df_series_clean.sample(sample_size).replace(to_replace=r, value='').map(len)

# Determine the max and min precision                                              
min_precision = float(len_per_float.min())
max_precision = float(len_per_float.max())

return min_precision
```